### PR TITLE
Document testing of discovery-ec2 and repository-s3.

### PR DESCRIPTION
**Pull request by dblock**
### Description

This is a start for documenting testing for discovery-ec2 and repository-s3.

### Related Issues

- https://github.com/opensearch-project/OpenSearch/pull/7372#issuecomment-1570531667
- https://github.com/opensearch-project/OpenSearch/issues/7867
- https://github.com/opensearch-project/OpenSearch/issues/7866
- https://github.com/opensearch-project/OpenSearch/pull/7974

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by dblock** [comment]
@raghuvanshraj and @dreamer-89 care to contribute to this?

File: plugins/repository-s3/README.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# repository-s3
```
Comment (dreamer-89) [inline plugins/repository-s3/README.md:1]: Thank you @dblock for taking effort on adding this documentation. Care to add same for other similar plugins e.g. repository-gcs, repository-azure, repository-hdfs ?

File: plugins/repository-s3/README.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+### Unit Tests
+
+```
+./gradlew :plugins:repository-s3:test
+```
+
+### Integration Tests
+
+Integration tests require `amazon_s3_secret_key` and `amazon_s3_bucket` environment variables.
```
Comment (dreamer-89) [inline plugins/repository-s3/README.md:15]: We also need to provide `amazon_s3_access_key`, `amazon_s3_base_path` as mentioned in sample command below.

File: plugins/repository-s3/README.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+## Testing
+
+### Unit Tests
+
+```
+./gradlew :plugins:repository-s3:test
+```
+
+### Integration Tests
```
Comment (dreamer-89) [inline plugins/repository-s3/README.md:13]: 1. Update this to share that setting up these vars when using actual s3 end-point is desired. Use this when there are changes in s3-plugin.
2. Add another section when user doesn't care about using actual s3.
```
 If using actual s3 service is not desired, we can skip setting up these vars. The plugin defaults to dummy  s3 service (using fixtures) which does not need above environment variables. 

./gradlew :plugins:repository-s3:s3ThirdPartyTest
```

File: plugins/repository-s3/README.md (hunk 1 above)
Comment (dblock) [reply plugins/repository-s3/README.md:1]: I really don't know anything about the other repository plugins other than at a high level, would prefer if the next person to work on them dug up the details and documented them.

File: plugins/repository-s3/README.md (hunk 3 above)
Comment (dblock) [reply plugins/repository-s3/README.md:13]: So this is no longer working and is https://github.com/opensearch-project/OpenSearch/issues/7867. I've added a note there to add to the documentation when we fix it.

File: plugins/repository-s3/README.md (hunk 2 above)
Comment (dblock) [reply plugins/repository-s3/README.md]: Added.

**Review by dblock — COMMENTED** [review]
@dreamer-89 Thanks for the review. Updated. Care to take another look? I'd like to err on the side of progress vs. perfection - we'll keep adding to these incrementally.

**Review by dreamer-89 — APPROVED** [review]
LGTM! 
Let's get one more review on discovery-ec2 doc.

**Comment by dreamer-89** [comment]
> ### Gradle Check (Jenkins) Run Completed with:
> * **RESULT:** FAILURE ❌
> * **URL:** https://build.ci.opensearch.org/job/gradle-check/17231/
> * **CommitID:** [2ac243b](https://github.com/opensearch-project/OpenSearch/commit/2ac243bb28812fdbe06acaa5c35b4567ebc58694)
>   Please examine the workflow log, locate, and copy-paste the failure(s) below, then iterate to green.
>   Is the failure [a flaky test](https://github.com/opensearch-project/OpenSearch/blob/main/DEVELOPER_GUIDE.md#flaky-tests) unrelated to your change?

@dblock : Need spotless fix on changes. 
```
* What went wrong:
Execution failed for task ':plugins:repository-s3:spotlessMiscCheck'.
> The following files had format violations:
      README.md
          @@ -22,4 +22,4 @@
           
           ```
           AWS_REGION=us-west-2·amazon_s3_access_key=$AWS_ACCESS_KEY_ID·amazon_s3_secret_key=$AWS_SECRET_ACCESS_KEY·amazon_s3_****_path=path·amazon_s3_bucket=dblock-opensearch·./gradlew·:plugins:repository-s3:s3ThirdPartyTest
          -```
          +```
  Run './gradlew :plugins:repository-s3:spotlessApply' to fix these violations.
```

**Comment by dblock** [comment]
Codecov is wrong. There's no code changes here.

**Comment by dreamer-89** [comment]
Tagging @raghuvanshraj @andrross @mch2  for review

**Pull request merged**